pkg/envoy_access_log_server/service_v3.go

```diff
@@ -0,0 +1,225 @@
+package envoy_access_log_server
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"time"
+
+	envoy_data_accesslog_v3 "github.com/envoyproxy/go-control-plane/envoy/data/accesslog/v3"
+	envoy_service_accesslog_v3 "github.com/envoyproxy/go-control-plane/envoy/service/accesslog/v3"
+	"github.com/golang/protobuf/ptypes"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+
+	"github.com/seznam/slo-exporter/pkg/event"
+	"github.com/seznam/slo-exporter/pkg/stringmap"
+)
+
+type AccessLogServiceV3 struct {
+	outChan chan *event.Raw
+	logger  logrus.FieldLogger
+	envoy_service_accesslog_v3.UnimplementedAccessLogServiceServer
+}
+
+type envoyV3AccessLogEntryCommonProperties envoy_data_accesslog_v3.AccessLogCommon
```

Comment:
Maybe we misunderstood somehow. I did not mean having a separate type for each of the log entry part, just for the whole log entry. Not sure if this brings anything compared to the separate functions. I’d have just the `envoyV3HttpAccessLogEntry` and `envoyV3TcpAccessLogEntry` and those with the `ToStringMap()` function. 
That's the only thing we care about with them actually.

Reply:
One of the @lksv's requirements was to separate request and response extraction to a separate, tested methods.

With that in mind, it made sense to me to split it as it is now.